app/livechat/client/views/app/tabbar/customerChatHistory.js

```diff
@@ -0,0 +1,230 @@
+	},
+	
+	previousChats() {
+		// will return pervious chats list
+		let history = Template.instance().history.get();
+		let newHisTory = []
+		for(i=1; i<history.length; i++){
+			var d = Date.parse(history[i].ts);
+			var D = new Date();
+			D.setTime(d);
+			var newTs = D.toTimeString();
+			var time = getTime(newTs);
+			history[i].time = time;
+			history[i].count = history[i].msgs-3;
+			newHisTory.push(history[i]);
+		}
+		Template.instance().modifiedHistory.set(newHisTory);
+		return Template.instance().modifiedHistory.get();
+	},
+	len(){
+		// will return length of total messages in room
+		len = Template.instance().historyResu.get();
+		return len = len.length-2;
+	},
+	title() {
+		
+		let title = moment(this.ts).format('L LTS');
+
+		if (this.label) {
+			title += ` - ${ this.label }`;
+		}
+		return title;
+	},
+});
+
+Template.customerChatHistory.onCreated(function() {
+	const currentData = Template.currentData();
+	this.visitorId = new ReactiveVar();
+	this.isLoading = new ReactiveVar(false);
+	this.modifiedHistory = new ReactiveVar([])
+	this.history = new ReactiveVar([]);
+	this.offset = new ReactiveVar(0);
+	this.total = new ReactiveVar(0);
+	this.isAllChat = new ReactiveVar(true);
+	this.isChatClicked = new ReactiveVar(true);
+	this.autorun(async () => {
+		const { room } = await APIClient.v1.get(`rooms.info?roomId=${ currentData.rid }`);
+		if (room && room.v) {
+			this.visitorId.set(room.v._id);
+		}
+	});
+
+	this.autorun(async () => {
+		
+		if (!this.visitorId.get() || !currentData || !currentData.rid) {
+			return;
+		}
+
+		const offset = this.offset.get();
+	
+		this.isLoading.set(true);
+		const { history, total } = await APIClient.v1.get(`livechat/visitors.chatHistory/room/${ currentData.rid }/visitor/${ this.visitorId.get() }?count=${ ITEMS_COUNT }&offset=${ offset }`);
+		
+		this.isLoading.set(false);
+		this.total.set(total);
+		this.history.set(this.history.get().concat(history));
+		// this will load all the chat history chat messages
+		allHistoryChat = [];
+		var limit = 100;
+		let count = this.history.get();
+		if(allHistoryChat.length>0){
+			return
+		}
+		for(i=0;i<count.length;i++){
+			let id = count[i]._id;
+			let token = count[i].v.token;
+			setTimeout(async function(){
+			const historyResult  =  await APIClient.v1.get(`livechat/messages.history/${ id }?token=${token}&limit=${limit}`);
+			let count2 = historyResult.messages;
+			
+			for(j=0; j<count2.length; j++){
+
+					allHistoryChat.push(count2[j]);
+				
+			}
+			
+		},1000)
+			
+		}
+		
+	
+	});	
+});
+
+Template.customerChatHistory.events({
+	'scroll .visitor-scroll': _.throttle(function(e, instance) {
+		if (e.target.scrollTop >= (e.target.scrollHeight - e.target.clientHeight)) {
+			const history = instance.history.get();
+			if (instance.total.get() <= history.length) {
+				return;
+			}
+			return instance.offset.set(instance.offset.get() + ITEMS_COUNT);
+		}
+	}, 200),
+	'keyup #searchInput': async function(event,template){ 
```

Comment:
You need to filter the rooms on the server-side which means you need to have a specific method to load the customer history chats and here you will use the same method, but you will pass the search term to the server and get the results.

Also, you need to use `debounce` to filter the chat history. We have a very similar use case here:
https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/client/views/app/livechatDepartments.js#L81

Please, take a look at that code and use it as inspiration.

Reply:
now i am using the search api to search the message inside a room